src/components/popover-container/popover-container.component.js

```diff
@@ -12,35 +11,20 @@ import {
 import Icon from '../icon';
 
 const PopoverContainer = ({
-  children, iconType, title, position
+  children, title, position, isOpen, onClose, renderOpenComponent, isStickyTop
 }) => {
-  const [isOpen, setOpen] = useState(false);
-  const iconRef = useRef();
   const closeIconRef = useRef();
 
   useEffect(() => {
-    if (isOpen) closeIconRef.current.focus();
-  }, [isOpen]);
-
-  const handleOpen = () => {
-    setOpen(true);
-  };
-
-  const handleClose = () => {
-    setOpen(false);
-    iconRef.current.focus();
-  };
+    if (isOpen && onClose) closeIconRef.current.focus();
+  }, [isOpen, onClose]);
 
   return (
     <PopoverContainerWrapperStyle data-component='popover-container'>
-      <PopoverContainerIcon
-        ref={ iconRef }
-        data-element='popover-container-icon'
-        tabIndex={ isOpen ? -1 : 0 }
-        onAction={ handleOpen }
-      >
-        <Icon type={ iconType } />
-      </PopoverContainerIcon>
+      {React.cloneElement(renderOpenComponent(), {
```

Comment:
Rather than `cloneElement` I think it would be better to use a render prop to achieve this. I'm currently updating `ActionPopover` to support this same behaviour.
```
<ActionPopover renderButton={ ({tabIndex, dataElement, styleOverride}) => {
return (
<CustomComponent
tabIndex={tabIndex}
'data-element'={dataElement}
styleOverride={styleOverride}
>Foo</CustomComponent>
);
}
>...</ActionPopover>
```